### DISCUSSION TITLE
Jinja2 templates

### DISCUSSION THREAD

**Discussion by pupdike** [adieyal/sd-dynamic-prompts#232] [General]
"If you are using these templates, please let me know if they are useful."

Just wanted to let you know the use case I found for them.  When I create an embedding I normally create an output about every 100 steps.  If I train only 1000 steps that means 10 unique embeddings to test.

Using this code I can test 10 of them relatively quickly:

{% for number in ['100', '200', '300', '400', '500', '600', '700', '800', '900', '1000'] %}
        {% prompt %}YOUR-EMBEDDING-NAME-{{ number}}:1.0{% endprompt %}
    {% endfor %}

I get the grid image which has all of the images in it and I can quickly see the progress of the learning and spot under/overlearning and take action based upon it.

**Comment by pupdike** [comment]
Here is a better version that allows repeating each embedding 4 times in a row (with an embedding weight of 1.0.)

{% for number in range(100, 1001, 100) %}
    {% for iteration in range(0, 4, 1) %}
        {% prompt %}YOUR-EMBEDDING-NAME-{{ number}}:1.0{% endprompt %}
    {% endfor %}
{% endfor %}

**Comment by adieyal** [comment]
great use case, thank you
